snuba/admin/static/ratelimit/index.tsx

```diff
@@ -0,0 +1,426 @@
+
+import { Table } from "../table";
+import Client from "../api_client";
+import {
+  Config,
+  ConfigKey,
+  ConfigValue,
+  ConfigType,
+  ConfigDescription,
+  ConfigDescriptions,
+  RowData,
+} from "./types";
+import { getEditableRow, getReadonlyRow, getNewRow } from "./row_data";
+import {
+  containerStyle,
+  defaultOptionStyle,
+  linkStyle,
+  paragraphStyle,
+} from "./styles";
+
+function RuntimeConfig(props: { api: Client }) {
+  const { api } = props;
+
+  // Original data from the API
+  const [data, setData] = useState<
+    | {
+        key: ConfigKey;
+        value: ConfigValue;
+        description: ConfigDescription;
+        type: ConfigType;
+      }[]
+    | null
+  >(null);
+
+  // Data after column filters are applied (dynamically changes)
+  const [filteredData, setSearchFilteredData] = useState<
+    | {
+        key: ConfigKey;
+        value: ConfigValue;
+        description: ConfigDescription;
+        type: ConfigType;
+      }[]
+    | null
+  >(null);
+
+  // All descriptions from API
+  const [allDescriptions, setDescriptions] = useState<ConfigDescriptions>({});
+
+  // Key of existing row being edited (if any)
+  const [currentlyEditing, setCurrentlyEditing] = useState<ConfigKey | null>(
+    null
+  );
+
+  // True if we are adding a brand new config, otherwise false
+  const [addingNew, setAddingNew] = useState(false);
+
+  // Unsaved state of the row currently being edited
+  const [currentRowData, setCurrentRowData] = useState<{
+    key: ConfigKey;
+    value: ConfigValue;
+    description: ConfigDescription;
+    type: ConfigType;
+  }>({ key: "", value: "", description: "", type: "string" });
+
+  // All rate limit dimensions, parameters, and targets that are selectable on the filter dropdowns
+  const [selectableDimensions, setSelectableDimensions] = useState<
+    Array<string>
+  >([]);
+  const [selectableParameters, setSelectableParameters] = useState<
+    Array<string>
+  >([]);
+  const [selectableTargets, setSelectableTargets] = useState<Array<string>>([]);
+
+  // The current applied filters
+  const [filterTerm, setFilterTerms] = useState<any>({
+    dimension: "",
+    parameter: "",
+    target: "",
+  });
+
+  // All supported rate limiter dimensions and parameters in Snuba
+  const availableDimensions = [
+    "project_referrer",
+    "project",
+    "referrer",
+    "organization",
+  ];
+  const availableParameters = ["per_second_limit", "concurrent_limit"];
+
+  function resetCurrentRowData() {
+    setCurrentRowData({ key: "", value: "", description: "", type: "string" });
+  }
+
+  // Load data if it was not previously loaded
+  if (data === null) {
+    fetchData();
+  }
+
+  function fetchData() {
+    api.getConfigs().then((res) => {
+      let data = filterOnlyRateLimitConfigs(res);
+      populateFilterLists(data);
+      setSearchFilteredData(data);
+      setData(data);
+    });
+  }
+
+  function filterOnlyRateLimitConfigs(configs: Array<Config>) {
```

Comment:
I thought we had talked about moving this functionality into the Python backend and either pass a parameter to the config endpoint to only return rate limiting or have a separate endpoint altogether?

Reply:
We have, this was just the original PR. Probably should have been a draft. Those backend changes are still being made.